# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../udev/SC_GET_SIZE_MAX.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- udev-143/libudev/libudev-util-private.c.vanilla	2011-01-05 16:43:47.000000000 +0100
+++ udev-143/libudev/libudev-util-private.c	2011-01-05 16:44:11.000000000 +0100
@@ -20,6 +20,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <sys/wait.h>
+#include <limits.h>
 
 #include "libudev.h"
 #include "libudev-private.h"
@@ -122,7 +122,7 @@
 uid_t util_lookup_user(struct udev *udev, const char *user)
 {
 	char *endptr;
-	int buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
+	int buflen = PATH_MAX; /* sysconf(_SC_GETPW_R_SIZE_MAX); */
 	char buf[buflen];
 	struct passwd pwbuf;
 	struct passwd *pw;
@@ -148,7 +148,7 @@
 gid_t util_lookup_group(struct udev *udev, const char *group)
 {
 	char *endptr;
-	int buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
+	int buflen = PATH_MAX; /* sysconf(_SC_GETGR_R_SIZE_MAX); */
 	char buf[buflen];
 	struct group grbuf;
 	struct group *gr;
